src/CachedKeySet.php

```diff
@@ -178,6 +178,12 @@ private function keyIdExists(string $keyId): bool
             }
             $request = $this->httpFactory->createRequest('GET', $this->jwksUri);
             $jwksResponse = $this->httpClient->sendRequest($request);
+            if ($jwksResponse->getStatusCode() !== 200) {
+                throw new UnexpectedValueException(
+                    (string) $jwksResponse->getBody(),
```

Comment:
Simple and cool. I'd be worried though, that any wrong response might respond with a large body. Or some weird JSON (do we even set a specific Accept header?).
As we have an PSR client though, we could use the simple:
```suggestion
                    $jwksResponse->getReasonPhrase(),
```
Or we could use a specific exception message like the rest of the library does.

Reply:
The Reason Phrase would be something like "OK" for 200, or "Bad Request" for 400. This may be what you think would be best, but it would potentially lose critical information in the request body that details the actual error being thrown here.

Maybe a better solution would be something like: 

```php
$responseBody = (string) $jwksResponse->getBody();
$exceptionMessage = $jwksResponse->getReasonPhrase() . ($responseBody? . ":\n" . $responseBody : '');
throw new UnexpectedValueException($exceptionMessage, $jwksResponse->getStatusCode());
```

Reply:
Would it be acceptable to truncate the response to a maximum length (say, 1000 bytes)? It does add some complexity to the code, but allows the response body to be exposed in a safer way.

Reply:
Well since PSR-7 must not throw exceptions it's pretty much up to wrapping implementations.

We could simplify the Guzzle way like this:

```suggestion
                    sprintf(
                        "Request Error: `GET %s` resulted in a `%s %s` response:\n%s",
                        $this->jwksUri,
                        $jwksResponse->getStatusCode(),
                        $jwksResponse->getReasonPhrase(),
                        mb_substr($jwksResponse->getBody(), 0, 1000)
                    ),
```

Here is what i had in my own wrapper since i found the order easier to read and the body not useful:
```suggestion
                    sprintf('Got bad HTTP status code %d %s for URI "%s"',
                        $jwksResponse->getStatusCode(),
                        $jwksResponse->getReasonPhrase(),
                        $this->jwksUri,
                    ),
```

I have a good argument for the body not being useful. The JWKS is a public resource so you can pop the URI into a browser and see yourself most of the time what is happening there. Or you can curl it on the server to really debug.